Kinvey/Kinvey/Result.swift

```diff
@@ -8,9 +8,13 @@
 
 import Foundation
 
+/// Enumeration that represents a result expected usually after an async call
```

Comment:
Also, this might be the right place to show how to use this enum. Do you think we should describe that?